core/trino-main/src/main/java/io/trino/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -181,21 +183,27 @@ private AggregationAnalyzer(
         // No defensive copy here for performance reasons.
         // Copying this map may lead to quadratic time complexity
         this.columnReferences = analysis.getColumnReferenceFields();
+    }
 
-        // Normal iteration instead of stream for performance reasons
-        ImmutableSet.Builder<FieldId> groupingFieldsBuilder = ImmutableSet.builder();
-        for (Expression expression : groupByExpressions) {
-            ResolvedField resolvedField = columnReferences.get(NodeRef.of(expression));
-            if (resolvedField != null) {
-                groupingFieldsBuilder.add(resolvedField.getFieldId());
+    private Set<FieldId> getGroupingFields()
+    {
+        if (groupingFields == null) {
```

Comment:
When we evaluate them lazily would there be a chance - where we won't use the `groupingFIelds` information ?

Reply:
The only invocations are `getGroupingFields().contains(fieldId)` so it is always used, after init.